contracts/ExampleOracleSimple.sol

```diff
@@ -30,30 +31,28 @@ contract ExampleOracleSimple {
         uint112 reserve0;
         uint112 reserve1;
         (reserve0, reserve1, blockTimestampLast) = _pair.getReserves();
-        assert(reserve0 != 0 && reserve1 != 0); // ensure that there's liquidity in the pair
-        assert(blockTimestampLast != 0); // ensure there's a price history
+        require(reserve0 != 0 && reserve1 != 0, 'ExampleOracleSimple: NO_RESERVES'); // ensure that there's liquidity in the pair
+        require(blockTimestampLast != 0, 'ExampleOracleSimple: NO_PRICE_HISTORY'); // ensure there's a price history
     }
 
     function update() external {
+        // updates the block timestamp and reserves so we don't have to adjust for it
+        pair.sync();
```

Comment:
my preference is to have at least one oracle example which does the counterfactual math, because it can be a significant gas savings. plus, more reason to use the nice FixedPoint functions!

we can throw a comment disclaimer in though, and also use the `sync` strategy in other oracles.